packages/core/src/components/dialog/step.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from "classnames";
+import * as React from "react";
+import { polyfill } from "react-lifecycles-compat";
+
+import { AbstractPureComponent2, Classes } from "../../common";
+import { DISPLAYNAME_PREFIX, HTMLDivProps, IProps } from "../../common/props";
+
+export type StepId = string | number;
+
+export interface IMultistepDialogPanelProps {
+    updateDialog: (state: { enableNextButton: boolean }) => void;
+}
+
+export interface IStepProps extends IProps, Omit<HTMLDivProps, "id" | "title" | "onClick"> {
+    /**
+     * Unique identifier used to control which step is selected.
+     */
+    id: StepId;
+
+    /**
+     * Indicates whether the next button should be enabled by default on this step
+     */
+    nextButtonEnabledByDefault?: boolean;
```

Comment:
Wondering if we should just expose `nextButtonProps` on `MultistepDialog` and make the next button fully controlled. Then we wouldn't need `nextButtonEnabledByDefault` or `updateDialog`, and `Step` would be a purely presentational component that could just render its children. If the next button needs to be disabled at the start of certain steps, the client would do that in their change handler for MultistepDialog's `onChange`. Since the parent component of `MultistepDialog` is ultimately responsible for keeping track of what data has been entered in the dialog, it might make sense to just allow it to control when the next button is enabled as well.

Reply:
Hmm, maybe we can expose a enableNext prop in the MultistepDialog instead of exposing nextButtonProps. I don't think that the user should be able to control any aspect of this button except its enabled state. If the user could change the button design such as the text, intent, etc. it would enable them to deviate from the Blueprint design, and I can't think of a case where this would be desirable (since we want to standardize this). For instance, enabling them to change a button on step 3 out of 5 to show "submit" would break the flow and cause confusion.

Reply:
Yeah! My bad, I missed that. I fixed it in the latest revision.